app/models/manageiq/providers/amazon/cloud_manager/event_catcher/runner.rb

```diff
@@ -42,7 +24,23 @@ def process_event(event)
     end
   end
 
+  private
+
   def filtered?(event)
     filtered_events.include?(event["messageType"])
   end
+
+  def event_monitor_handle
+    @event_monitor_handle ||= begin
+      stream = ManageIQ::Providers::Amazon::CloudManager::EventCatcher::Stream.new(@ems)
+      stream.before_poll do
+        heartbeat
```

Comment:
I'm doing a heartbeat every 20 seconds, otherwise the worker would be restarted over and over in case there are no events coming from amazon